### PR TITLE
Use the interpreter to run the wheel resolver

### PR DESCRIPTION
**Pull request by peterebden**
The resolver just has `/usr/bin/env python3` (more or less) which might not necessarily be the Python we want to use (and this is important because it uses that Python to figure out arch tags etc). This invokes it more specifically so it honours the config settings.

### CODE REVIEW

**Review by toastwaffle — APPROVED** [review]

File: build_defs/python.build_defs [hunk 1]
```diff
@@ -513,9 +513,9 @@ def python_wheel(name:str, version:str, labels:list=[], hashes:list=None, packag
                  outs:list=None, post_install_commands:list=None, patch:str|list=None, licences:list=None,
                  test_only:bool&testonly=False, repo:str=None, zip_safe:bool=True, visibility:list=None,
                  deps:list=[], name_scheme:str|list=CONFIG.PYTHON.WHEEL_NAME_SCHEME,
-                 strip:list=['*.pyc', 'tests'], binary = False, entry_points={},
+                 strip:list=['*.pyc', 'tests'], binary = False, entry_points:dict={},
                  tool:str=CONFIG.PYTHON.WHEEL_TOOL, prereleases:bool=CONFIG.PYTHON.PRERELEASES,
-                 tool_verbosity:str=CONFIG.PYTHON.VERBOSITY):
+                 tool_verbosity:str=CONFIG.PYTHON.VERBOSITY, interpreter:str=CONFIG.PYTHON_DEFAULT_INTERPRETER):
```
Comment (chrisnovakovic) [inline build_defs/python.build_defs:518]:
```suggestion
                 tool_verbosity:str=CONFIG.PYTHON.VERBOSITY, interpreter:str=CONFIG.PYTHON.DEFAULT_INTERPRETER):
```

Comment (peterebden) [reply build_defs/python.build_defs]: yup thanks, every single test failed which also pointed to this :)

File: build_defs/python.build_defs [hunk 2]
```diff
@@ -513,9 +513,9 @@ def python_wheel(name:str, version:str, labels:list=[], hashes:list=None, packag
                  outs:list=None, post_install_commands:list=None, patch:str|list=None, licences:list=None,
                  test_only:bool&testonly=False, repo:str=None, zip_safe:bool=True, visibility:list=None,
                  deps:list=[], name_scheme:str|list=CONFIG.PYTHON.WHEEL_NAME_SCHEME,
-                 strip:list=['*.pyc', 'tests'], binary = False, entry_points={},
+                 strip:list=['*.pyc', 'tests'], binary = False, entry_points:dict|str={},
```
Comment (peterebden) [inline build_defs/python.build_defs:516]: this was missing before, but was documented as such

File: third_party/python/BUILD [hunk 3]
```diff
@@ -453,7 +453,7 @@ python_wheel(
 
 python_wheel(
     name = "tomli",
-    hashes = ["sha256: 939de3e7a6161af0c887ef91b7d41a53e7c5a1ca976325f429cb46ea9bc30ecc"],
```
Comment (peterebden) [inline third_party/python/BUILD:456]: this was emitting a warning. it'd be fine without the space, but the others don't have the prefix anyway.

**Pull request merged**